source/Ship.cpp

```diff
@@ -2305,7 +2328,40 @@ void Ship::Launch(list<shared_ptr<Ship>> &ships, vector<Visual> &visuals)
 			// Resupply any ships launching of their own accord.
 			if(!ejecting)
 			{
-				// TODO: Restock fighter weaponry that needs ammo.
+				// Determine which of the fighter's weapons we can restock.
+				auto restockable = bay.ship->GetArmament().RestockableAmmo();
+				bool tookAmmo = false;
+				// Ammunition found in the cargo hold can be transferred to internally carried fighters.
```

Comment:
Nice addition, but why in this PR? Can we just keep this for a separate follow-up PR?
(This is not just a code-change, but this is also relevant for game-balancing and content-creators. I would really like to keep the relevant discussions on those topics for a follow-up PR.)

Reply:
This was a nod to the functionality request in the sister PR, which was as popular as, and perhaps even more so, than this approach to the functionality. I included it in this PR, as the purpose of this PR is to add the ability to restock fighter ammunition from the carrier. Doing so via ammo-from-cargo, or ammo-from-outfits, is an implementation detail.

I don't remember exactly which was first (these PRs or the functionality), but content creators can design AI fleets that will make use of the ammo-from-cargo, through the use of the fleet "outfitters" syntax:

```
fleet "Carriers w/Cargo Ammo"
  outfitters "Ammo North" "Ammo South"
  ...
```

Reply:
`if(bay.side == Bay::INSIDE` is not an implementation detail, but an explicit choice about some bays being different than others. Most content-creators will not review source-code-changes into the level of detail in which they would pick this up and I would really like to give content-creators the chance to respond to this.

Another thought is that I still like to explore some other options for "ammo loading from cargo" as was suggested in the sister-PR, but without the downsides as were listed in that sister-PR. Think of outfits that get installed in the cargo space (not in the outfit space) and that:
- Very slowly load their ammo into the main Outfits/weapons
- Or only load ammo into fighters
- Or can only be installed on some carriers due to some attributes.

Adding the "from cargo for internal bays" functionality here will prevent exploration of some of those other options.